pkg/plugins/diff/delta_diff_server/src/main.rs

```diff
@@ -0,0 +1,211 @@
+use std::collections::HashMap;
+use std::fmt::Error;
+use std::io::{self, Write};
+use std::net::SocketAddr;
+
+use deltalake::delta::DeltaTable;
+use deltalake::DeltaDataTypeVersion;
+use portpicker;
+use serde_json::{Map, Value};
+use tonic::{Code, Request, Response, Status, transport::Server};
+
+use differ::{DiffRequest, DiffResponse, GatewayConfig, table_differ_server::{TableDiffer, TableDifferServer}};
+
+use crate::delta_ops::OpTypes;
+use crate::differ::{DiffProps, HistoryRequest, HistoryResponse, TableOperation, TablePath};
+
+mod delta_ops;
+mod utils;
+
+pub mod differ {
+    include!("diff.rs");
+}
+
+impl std::hash::Hash for TableOperation {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.id.hash(state);
+        self.operation.hash(state);
+        self.operation_type.hash(state);
+        self.content.hasher();
+        self.timestamp.hash(state);
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct DifferService {}
+
+#[tonic::async_trait]
+impl TableDiffer for DifferService {
+    async fn table_diff(&self, request: Request<DiffRequest>) -> Result<Response<DiffResponse>, Status> {
+        let r = request.into_inner();
+        let s3_gateway_config_req: GatewayConfig = r.gateway_config.unwrap();
+        let s3_config_map: HashMap<String, String> = utils::construct_storage_config(s3_gateway_config_req);
+
+        let diff_props: DiffProps = r.props.unwrap();
+
+        let left_table_path: TablePath = diff_props.left_table_path.unwrap();
+        let right_table_path: TablePath = diff_props.right_table_path.unwrap();
+        // let base_table_path: Option<TablePath> = diff_props.base_table_path;
+        let left_table_res: Result<DeltaTable, Status> =
+            delta_ops::get_delta_table(&s3_config_map, &diff_props.repo, left_table_path).await;
+        let right_table_res: Result<DeltaTable, Status> =
+            delta_ops::get_delta_table(&s3_config_map, &diff_props.repo, right_table_path).await;
+
+        // Both tables not found
+        if right_table_res.is_err() && left_table_res.is_err() {
+            return Err(Status::new(Code::NotFound,
+                                   format!("{:?}, {:?}", left_table_res.unwrap_err(),
+                                           right_table_res.unwrap_err())
+            ));
+        }
+        // If the table exists on the left ref but doesn't exist on the right ref -> table dropped.
+        if right_table_res.is_err() && left_table_res.is_ok() {
+            return Ok(Response::new(DiffResponse { entries: vec![], diff_type: differ::DiffType::Dropped as i32 }));
```

Comment:
Same as @talSofer mentioned below. If the base here doesn't have that table too, it means that the left side created a table and not that the right side deleted it.